build.gradle

```diff
@@ -40,17 +41,18 @@ task sourcesJar(type: Jar) {
 
 tasks.withType(JavaCompile) {
     Provider<JavaCompiler> compilerProvider = javaToolchains.compilerFor {
-        JavaLanguageVersion.of(8)
+        JavaLanguageVersion.of(16)
     }
     JavaCompiler compiler = compilerProvider.orNull
 
-    if (compiler != null && compiler.metadata.languageVersion.asInt() > 9) {
+    // TODO maybe adjust this comments and logic
+    if (compiler != null && compiler.metadata.languageVersion.asInt() > 16) {
         // Mark release as 8 if the compiler is Java 10 or newer
-        options.release.set(8)
+        options.release.set(16)
     } else {
         // Only use the legacy compiler options otherwise
-        sourceCompatibility = 1.8
-        targetCompatibility = 1.8
+        sourceCompatibility = 16
+        targetCompatibility = 16
     }
```

Comment:
Do we even need this logic here? Since an older (legacy) compiler won't even be able to compile this into Java16 bytecode, right?

Perhaps @md5sha256 knows better.

Reply:
Ya that's why I've added a todo for me because it shouldn't be necessary anymore or mostly just somehow adjusted.

Reply:
We won't need source and target here, release replaces those two legacy options. And yes, as Biscuit mentions, we won't be able to compile Java 16 bytecode if we're using a compiler which wants source and target as opposed to release.

Reply:
So I would remove the complete method and just add the `options.release.set(...)` thingy. Or do we want to remove this here completely? It would just be there as an indicator that we use Java 16+.

Reply:
I think what @md5sha256 tried to say is that we only need `options.release.set(16)` here and can remove the rest, if I understoof correctly.
I am relying on him here as I don't use gradle this often.

Reply:
I agree here with @md5sha256  and @TheBusyBiscuit , we should remove the if and simple have an `options.release.set(16)` here

Reply:
done